### PR TITLE
FlyToViewport: add support for auto duration

### LINKED ISSUE #96: FlyToViewport: add utility method to compute transition duraiton
This utility method is needed to support 'auto` duration mode for viewport fly-to transitions.

Ref: https://github.com/uber/react-map-gl/issues/830

### PR DESCRIPTION
**Pull request by 1chandu**
Fixes #96

### CODE REVIEW

File: docs/api-reference/web-mercator-utils.md [hunk 1]
```diff
@@ -201,3 +201,23 @@ Parameters:
 Returns:
 - `{longitude, latitude, zoom}`
 
+### `getFlyToDuration(startProps, endProps, opts)`
+
+Returns time in milliseconds, that is required perform transition from one viewport to another. Time returned is proportional to the distance we are transitioning to. This util function implements the algorithm described in “Smooth and efficient zooming and panning.” by Jarke J. van Wijk and Wim A.A. Nuij.
+
+Parameters:
+- `startProps` (Object) - viewport to fly from
+- `startProps.width` (Number, required)
```
Comment (Pessimistress) [inline docs/api-reference/web-mercator-utils.md]: Use nested list?

File: src/fly-to-viewport.js [hunk 2]
```diff
@@ -77,4 +58,45 @@ export default function flyToViewport(startProps, endProps, t) {
...
 }
-/* eslint-enable max-statements */
+
+// returns transition duration in milliseconds
+export function getFlyToDuration(startProps, endProps, opts = {}) {
+
+  const {S} = getFlyToTransitionParams(startProps, endProps);
+  const {speed} = opts;
+
+  return 1000 * S / (speed || DEFAULT_PROPS.speed);
```
Comment (Pessimistress) [inline src/fly-to-viewport.js]: What is the unit of `S`? Shall we just export a `getFlyToDistance` and let react-map-gl determine what options to expose?

Comment (1chandu) [reply src/fly-to-viewport.js]: `S` is an intermediate value produced based on zoom and position (lng/lat) change, there is no specific unit for this.

Comment (Pessimistress) [reply src/fly-to-viewport.js]: You either need to explain `S` or explain `speed`, which does not have more clarity...

File: docs/api-reference/web-mercator-utils.md (hunk 1 above)
Comment (1chandu) [reply docs/api-reference/web-mercator-utils.md]: Rest of the documents have the same style.

File: src/fly-to-viewport.js (hunk 2 above)
Comment (1chandu) [reply src/fly-to-viewport.js]: Updated to return `S`,  changed util method name to :`getFlyToLength`.

**Pull request merged**